src/cpp/src/block_manager.hpp

```diff
@@ -75,56 +78,81 @@ class KVCacheBlock {
 
 
 class Evictor {
-    std::map<size_t, KVCacheBlock::Ptr> blocks;
+    std::map<size_t, std::vector<KVCacheBlock::Ptr>> blocks;
+    size_t m_num_layers;
 public:
+    Evictor(size_t num_layers) : m_num_layers(num_layers) {}
```

Comment:
```suggestion
    explicit Evictor(size_t num_layers) : m_num_layers(num_layers) {}
```

Reply:
Done, also renamed the class to `BlockHashStore` so that there are no name collisions.